### PR TITLE
Implement some WOG Mage cards

### PR DESCRIPTION
**Pull request by astradamus**

### CODE REVIEW

**Comment by astradamus** [comment]
Should I test to make sure the deck is undisturbed by Cabalist's Tome (i.e. make sure we're not drawing)? Couldn't decide on that. I also put together a full hand test for Tome, but it ended up seeming pretty redundant since you can't resolve the effect with 10 cards in your hand so really it's just testing whether Give is properly protected.

**Comment by jleclanche** [comment]
Can you write a test for Shatter please? It's the first card to make use of `REQ_TARGET_FROZEN` which I believe isn't implemented right now.

**Comment by jleclanche** [comment]
> Should I test to make sure the deck is undisturbed by Cabalist's Tome

No need.

File: fireplace/cards/wog/mage.py [hunk 1]
```diff
@@ -9,6 +9,11 @@ class OG_083:
 	play = Hit(ENEMY_MINIONS, 1)
 
 
+class OG_085:
+	"Demented Frostcaller"
+	events = OWN_SPELL_PLAY.on(Freeze(RANDOM(ENEMY_CHARACTERS - MORTALLY_WOUNDED - FROZEN)))
```
Comment (jleclanche) [inline fireplace/cards/wog/mage.py]: You should be using `.after()`

File: tests/test_wog.py [hunk 2]
```diff
@@ -114,6 +114,23 @@ def test_chogall_cannot_pay_health():
...
+		game.player2.give(WISP).play()
+	game.end_turn()
+	enemies = game.player2.characters
+	# Cast 8 coins, ensuring enemies are frozen one by one.
+	for x in range(1,9):
+		game.player1.give(THE_COIN).play()
+		assert len([en for en in enemies if en.frozen]) == x
+	# Cast one extra coin, ensuring nothing breaks when all enemies are already frozen.
+	game.player1.give(THE_COIN).play()
+
```
Comment (jleclanche) [inline tests/test_wog.py]: You don't need such an extensive test. A single wisp with 3 coin casts (hero, wisp and null) is fine.

Also remember to add newlines after `end_turn()`.

Comment (astradamus) [inline tests/test_wog.py]: Okay, will fix this up tonight.

**Comment by astradamus** [comment]
Uh, what happened here with CI? I didn't really do anything different here that I can see, and the tests passed fine on my end... it's failing tests I didn't touch because of something called PowSubType? My commits don't touch anything related to that. Color me puzzled!

Also, re: Shatter, what exactly am I testing? Making sure it fails when I try to target a non-frozen minion? If it doesn't fail (i.e. your hunch that frozen target requirement is not implemented is correct), should I comment out that line with a TODO or something? (I've seen such a commented out line in Discover cards trying to test card options are of the right class, i.e. Ethereal Conjurer).

File: tests/test_wog.py [hunk 3]
```diff
@@ -114,6 +114,22 @@ def test_chogall_cannot_pay_health():
...
+	game.player1.give("OG_085").play()
+	game.end_turn()
+
+	game.player2.give(WISP).play()
+	game.end_turn()
+
+	enemies = game.player2.characters
+	for x in range(1, 3):
+		game.player1.give(THE_COIN).play()
+		assert len([en for en in enemies if en.frozen]) == x
```
Comment (jleclanche) [inline tests/test_wog.py]: You can do `assert len(game.player2.characters.filter(frozen=True)) == i` instead.

**Comment by jleclanche** [comment]
Rebase on master + small fix in the frostcaller test and gtm. :)

**Comment by astradamus** [comment]
Oh, what? I never saw these comments. On it.

File: tests/test_wog.py (hunk 3 above)
Comment (astradamus) [inline tests/test_wog.py]: For one minute you had me thrilled that Python collections had filter! Alas. It's good to know you've got your own.

**Comment by astradamus** [comment]
Okay, all done, sorry for the delay.

**Comment by astradamus** [comment]
Okay, all done, sorry for the delay.

**Pull request merged**